pkg/internal/enhancedmetrics/clients/clients.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2026 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package clients
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/prometheus-community/yet-another-cloudwatch-exporter/pkg/internal/enhancedmetrics/config"
+	"github.com/prometheus-community/yet-another-cloudwatch-exporter/pkg/model"
+)
+
+// separator is NULL byte (0x00)
+const separator = "\x00"
```

Comment:
Where is that used? Maybe add it do the documentation.

Reply:
Good point, I have extended the description here:

```
// This is an arbitrary choice of a character that is unlikely to appear in region names or role ARNs/ExternalIDs.
// It is used to separate different components in the client key in the map when generating unique keys for region-role combinations.
```